src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -260,7 +260,16 @@ public OptionSet Options
 
         internal void UpdateCurrentSolutionOnOptionsChanged()
         {
-            SetCurrentSolution(CurrentSolution.WithOptions(new SolutionOptionSet(_legacyOptions)));
+            // TODO: this should be locked as it is updating CurrentSolution.  However, that triggers a deadlock in options due to:
+            // https://github.com/dotnet/roslyn/issues/64681
+            //
+            // This should be fixable once SolutionCrawler is entirely removed.
+            // using (_serializationLock.DisposableWait())
+            {
+                SetCurrentSolution(CurrentSolution.WithOptions(new SolutionOptionSet(_legacyOptions)));
+
+                // TODO: Are we missing a call to RaiseWorkspaceChangedEventAsync here?
```

Comment:
This seems busted; honestly now that I think about it I was triaging a handful of bugs which were options may have not been working for analyzers, and maybe this is a cause?

Reply:
yes.  it seems quite busted.  i can update this PR to emit a SolutionChanged notification here.